### ISSUE TITLE
could we have a `Categorical` constructor that takes `AbstractVector`

### ISSUE THREAD

**Issue by tpapp** [JuliaStats/Distributions.jl#559]
eg something along the lines of 
```julia
Categorical{T <: Real}(v::AbstractVector{T}) = Categorical(collect(v))
```
that handles `SparseVector` etc.

The issue came up when I was simulating a Markov chain with a sparse transition matrix, making `Categorical`s out of the rows.

**Comment by andreasnoack** [comment]
I don't think that would be a problem but I would probably make it `Categorical{T <: Real}(v::AbstractVector{T}) = Categorical(Vector{T}(v))` to avoid the copy when possible.

**Comment by tpapp** [comment]
Either one is fine with me. Should I make a PR, or can someone with write access make this trivial change? This would allow me to remove the workaround from my own code.

**Comment by andreasnoack** [comment]
A PR would be great
